compiler-rt/test/sanitizer_common/TestCases/replace_dlopen_main_program_test.cpp

```diff
@@ -0,0 +1,29 @@
+// Test 'test_only_replace_dlopen_main_program' flag
+
+// RUN: %clangxx %s -o %t
+// RUN: env %tool_options='test_only_replace_dlopen_main_program=true' %run %t
+// RUN: env %tool_options='test_only_replace_dlopen_main_program=false' not %run %t
+// REQUIRES: glibc
```

Comment:
This this be in Linux folder then?

Reply:
with dladdr/readlink it should work on any posix?

Reply:
I moved it to the Linux folder. 'man dladdr' says it is "nonstandard GNU extensions that are also present on Solaris".